plugwise/helper.py

```diff
@@ -175,103 +211,112 @@ async def request(
-            location_members = set()
+            loc.name = location.find("name").text
+            loc.id = location.attrib["id"]
+            loc.types = set()
+            loc.members = set()
 
             # Group of appliances
             locator = ".//appliances/appliance"
             if location.find(locator) is not None:
                 for member in location.findall(locator):
-                    location_members.add(member.attrib["id"])
-
-            if location_name == "Home":
-                home_location = location_id
-                location_types.add("home")
+                    loc.members.add(member.attrib["id"])
 
-                for location_type in types_finder(location):
-                    location_types.add(location_type)
+            # Specials
+            loc = self.locations_specials(loc, location)
 
-            # Legacy P1 right location has 'services' filled
-            # test data has 5 for example
-            locator = ".//services"
-            if (
-                self._smile_legacy
-                and self.smile_type == "power"
-                and len(location.find(locator)) > 0
-            ):
-                # Override location name found to match
-                location_name = "Home"
-                home_location = location_id
-                location_types.add("home")
-                location_types.add("power")
-
-            self._loc_data[location_id] = {
-                "name": location_name,
-                "types": location_types,
-                "members": location_members,
+            self._loc_data[loc.id] = {
+                "name": loc.name,
+                "types": loc.types,
+                "members": loc.members,
             }
 
-        self._home_location = home_location
-
         return
 
+    def appliance_stretch_class_finder(self, appliance, appl):
+        """Determine class per stretch appliance."""
+        if self.stretch_v2 or self.stretch_v3:
+            locator = ".//services/electricity_point_meter"
+            mod_type = "electricity_point_meter"
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            if appl.model != "Group Switch":
+                appl.model = None
+            if module_data[2] is not None:
+                hw_version = module_data[2].replace("-", "")
+                appl.model = version_to_model(hw_version)
+            appl.fw = module_data[3]
+            return appl
+
+        if self.smile_type != "stretch" and "plug" in appl.types:
+            locator = ".//logs/point_log/electricity_point_meter"
+            mod_type = "electricity_point_meter"
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            appl.model = version_to_model(module_data[1])
+            appl.fw = module_data[3]
+            return appl
+
     def appliance_class_finder(self, appliance, appl):
```

Comment:
Please change the name to `appliance_info_finder`, the use of `class` in the name does not cover its purpose.